### PR TITLE
API work.

### PR DESCRIPTION
**Pull request by MichaelNorman**
Type and member summaries for new types. Button, Cell, Configuration, and ConstraintExpression.

### CODE REVIEW

File: docs/Xamarin.Forms.Core/Xamarin.Forms/Cell.xml [hunk 1]
```diff
@@ -207,7 +207,7 @@ Content = new TableView
       </ReturnValue>
       <MemberValue>40</MemberValue>
       <Docs>
-        <summary>To be added.</summary>
+        <summary>The default height of cells.</summary>
```
Comment (StephaneDelcroix) [inline docs/Xamarin.Forms.Core/Xamarin.Forms/Cell.xml]: the doc could say the default value is 40

File: docs/Xamarin.Forms.Core/Xamarin.Forms/Button+ButtonContentLayout+ImagePosition.xml [hunk 2]
```diff
@@ -52,7 +52,7 @@
         <ReturnType>Xamarin.Forms.Button+ButtonContentLayout+ImagePosition</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>To be added.</summary>
+        <summary>Indicates that the image will appear at the reght edge of the button.</summary>
```
Comment (samhouts) [inline docs/Xamarin.Forms.Core/Xamarin.Forms/Button+ButtonContentLayout+ImagePosition.xml]: right, not reght

File: docs/Xamarin.Forms.Core/Xamarin.Forms/Cell.xml [hunk 3]
```diff
@@ -554,7 +554,7 @@ Content = new TableView
       </ReturnValue>
       <Parameters />
       <Docs>
-        <summary>To be added.</summary>
+        <summary>This method is for internal use by platform renderers.</summary>
```
Comment (StephaneDelcroix) [inline docs/Xamarin.Forms.Core/Xamarin.Forms/Cell.xml]: this is explicit interface implementation. I don't think it should appear in the docs

File: docs/Xamarin.Forms.Core/Xamarin.Forms/Button+ButtonContentTypeConverter.xml [hunk 4]
```diff
@@ -42,7 +42,7 @@
       </Parameters>
       <Docs>
         <param name="value">To be added.</param>
-        <summary>To be added.</summary>
+        <summary>Conerts <paramref name="value" /> into a <see cref="T:Xamarin.Forms.Button+ButtonContentLayout" /> object.</summary>
```
Comment (samhouts) [inline docs/Xamarin.Forms.Core/Xamarin.Forms/Button+ButtonContentTypeConverter.xml]: Converts, not Conerts

Comment (samhouts) [inline docs/Xamarin.Forms.Core/Xamarin.Forms/Button+ButtonContentTypeConverter.xml]: an object, not a object...?

**Pull request merged**